### PR TITLE
Freeze comments edited by someone else than original author

### LINKED ISSUE #321: One can edit others comments to gain approvals
I can edit others comments on my PR to make them +1 and PR approved.

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #321 

* Comment is persisted to database with (old) body if event equals `issue_comment.edited` or `issue_comment.deleted` and `sender.login !== comment.author.login`
* On checking approvals, frozen comments since the last push are fetched from the database and merged with the upstream comments, overwriting malicious edits.
* Frozen comments are deleted on `pull_request.synchronize`.

### CODE REVIEW

**Comment by prayerslayer** [comment]
Actually not so cool: With this logic you can still remove vetoes. Will change.

**Comment by neremic** [comment]
:+1:

**Comment by prayerslayer** [comment]
We could delete frozen comments after `pull_request.synchronize` as we are fetching them only since last push on PR anyways. Will do that later.

**Comment by neremic** [comment]
:+1:

File: test/server/github.test.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import { GithubService } from '../../server/service/GithubService'
+import {toGenericComment} from '../../common/util'
 
 describe('The Github service', () => {
   let github
   const MOCK_COMMENTS = [
-    {created_at: '2016-07-11T15:00:00Z'},
-    {created_at: '2016-07-11T17:00:00Z'}
+    {created_at: '2016-07-11T15:00:00Z', user: {login: 'foo'}},
+    {created_at: '2016-07-11T17:00:00Z', user: {login: 'bar'}}
```
Comment (taojang) [inline test/server/github.test.js]: ```user``` changed to string, or not?

Comment (prayerslayer) [inline test/server/github.test.js]: In this particular case the `user` is not important as we only test if the date filter works and check the length of the returned array. That's why I deleted it.

But yes... Comments from Github have `user` as an object. We convert them now to a generic object where `user` is a string (to handle them consistently with comments from the database).

Somebody should build `clojure.spec` in JS.

Comment (taojang) [inline test/server/github.test.js]: Thx for explanation :+1:

**Comment by taojang** [comment]
:+1:

**Pull request merged**